editor/editor_node.cpp

```diff
@@ -4406,6 +4411,71 @@ void EditorNode::progress_end_task_bg(const String &p_task) {
 	singleton->progress_hb->end_task(p_task);
 }
 
+String EditorNode::_get_system_info() const {
+	Vector<String> info;
+
+	// Fetch info.
+	String godot_version = String(VERSION_FULL_BUILD);
+
+	String distribution_name = OS::get_singleton()->get_distribution_name();
+	if (distribution_name.is_empty()) {
+		distribution_name = OS::get_singleton()->get_name();
+	}
+	if (distribution_name.is_empty()) {
+		distribution_name = "Other";
+	}
+	String distribution_version = OS::get_singleton()->get_version();
+
+	String driver_name = ProjectSettings::get_singleton()->get("rendering/driver/driver_name");
+	String rendering_device_name = RenderingServer::get_singleton()->get_rendering_device()->get_device_name();
+
+	RenderingDevice::DeviceType device_type = RenderingServer::get_singleton()->get_video_adapter_type();
+	String device_type_string;
+	switch (device_type) {
+		case RenderingDevice::DeviceType::DEVICE_TYPE_INTEGRATED_GPU:
+			device_type_string = "integrated";
+			break;
+		case RenderingDevice::DeviceType::DEVICE_TYPE_DISCRETE_GPU:
+			device_type_string = "dedicated";
+			break;
+		case RenderingDevice::DeviceType::DEVICE_TYPE_VIRTUAL_GPU:
+			device_type_string = "virtual";
+			break;
+		case RenderingDevice::DeviceType::DEVICE_TYPE_CPU:
+			device_type_string = "software emulation on CPU";
+			break;
+		case RenderingDevice::DeviceType::DEVICE_TYPE_OTHER:
+		case RenderingDevice::DeviceType::DEVICE_TYPE_MAX:
+			break; // Can't happen, but silences warning for DEVICE_TYPE_MAX
+	}
+
+	String processor_name = OS::get_singleton()->get_processor_name();
+	int processor_count = OS::get_singleton()->get_processor_count();
+
+	// Join info.
+	static String prefix = "*";
+	if (!distribution_version.is_empty()) {
+		info.push_back(vformat("%s OS: %s %s", prefix, distribution_name, distribution_version));
+	} else {
+		info.push_back(vformat("%s OS: %s", prefix, distribution_name));
+	}
+	info.push_back(vformat("%s Godot Version: %s", prefix, godot_version));
+	info.push_back(vformat("%s Rendering Driver: %s", prefix, driver_name));
+	if (device_type_string.is_empty()) {
+		info.push_back(vformat("%s Graphics Card: %s", prefix, rendering_device_name));
+	} else {
+		info.push_back(vformat("%s Graphics Card: %s (%s)", prefix, rendering_device_name, device_type_string));
+	}
+	info.push_back(vformat("%s CPU: %s (%d Cores)", prefix, processor_name, processor_count));
```

Comment:
`processor_count` is the number of logical CPU cores, which are generally called "threads".

```suggestion
	info.push_back(vformat("%s CPU: %s (%d Threads)", prefix, processor_name, processor_count));
```

Reply:
> On CPUs with HyperThreading enabled, this number will be greater than the number of physical CPU cores.

Ahhh, indeed. Good catch, thanks!